data/transactions/logic/eval.go

```diff
@@ -4006,48 +3988,71 @@ func (cx *EvalContext) accountReference(account stackValue) (basics.Address, uin
 	if err != nil {
 		return addr, 0, err
 	}
+
 	idx, err := cx.txn.Txn.IndexByAddress(addr, cx.txn.Txn.Sender)
+	if err == nil {
+		return addr, idx, nil
+	}
 
+	ok := cx.availableAccount(addr)
+	if !ok {
+		return addr, 0, err
+	}
 	invalidIndex := uint64(len(cx.txn.Txn.Accounts) + 1)
+	return addr, invalidIndex, nil
```

Comment:
Is it really invalid, more like faultIndex or accountsMissIndex

Reply:
I'll call it `nonLocalIndex` as it now implies we we don't have a "local" index into the `txn` itself, and must construct an index for it in SharedAccounts if we intend to modify it.

Reply:
Actually with code restructured as it is now, there's no point in using a variable at all.